ENH: Added MR modality specific attributes and improving usability by using argparse

che85 opened the pull request:
- using argparse for providing help and flagged parameters
- if dcmqi is required for an SR it will throw an exception if it was neither found in environment variable `DCMQI_PATH` nor as an additional parameter `-dcmqi <dcmqi binary path>`
- changed indention from 4 to 2 spaces in order to be consistent
- prompting user if output directory does not exist

che85 committed:
ENH: added MR modality table and additional information to CompositeContext

* using argparse, refined parameters
* dcmqi path can be defined in environment variable DCMQI_PATH or as an
  additional argument -dcmqi <dcmqi binary location>

che85 committed:
STYLE: to be consistent changed tab spaces to 2 instead of 4

che85 committed:
ENH: added SOPInstanceUID to MR specific attributes

che85 committed:
ENH: handle KeyError due to missing DICOM tag without quitting program

che85 committed:
ENH: updated usage and prompting user if output directory doesn't exist

* expecting input case insensitive [y, yes, true] or [n, no, false] from
  user when prompted regarding creation of non existing output directory

che85 committed:
BUG: missing message attribute in Exception object causes errors in python3

fedorov commented on schema.qdbd:
```diff
@@ -7,9 +7,12 @@
+ProductId
 SOPClassUID
 SOPInstanceUID PK
-StudyInstanceUID
```
This should not be removed, it is quite important.

fedorov commented on tabulate.py:
```diff
@@ -17,32 +20,65 @@
+        else:
+          print ("Make sure that you specified dcmqi path either in your environment variable 'DCMQI_PATH' or as an "
+                "additional parameter '-dcmqi <DCMQI binary path>'")
+        print ("Skipping %s" %dcmName)
```
I don't think it is a good idea to skip DICOM files when there is any of the attributes missing. They should be set to empty values, but parsing should continue for those files!

che85 replied:
I need to look into this before the PR is ready.

che85 committed:
ENH: added back StudyInstanceUID and created MR association with composite context

che85 committed:
BUG: added missing column FileName to Instance2File

- added slice thickness and pixel spacing to MR specific attributes
- removed unneeded parentheses

che85 committed:
ENH: trying to handle key errors more gentle without skipping DICOM files

fedorov commented:
@che85 what is the status of this PR?

che85 committed:
ENH: printing message to user if TID of SR is other than 1500

- which is not supported at the moment

che85 commented:
@fedorov I think https://github.com/QIICR/dcm2tables/pull/5/commits/d90a4fd6d8d68c2ac31c6559e74d5d55c029f47f already fixed your concerns regarding skipping of series. Now that only happens if a SR was found but no dcmqi executable is available. Or should it even in that situation just print a message and continue?

fedorov commented on DICOMParser.py:
```diff
@@ -55,6 +55,8 @@ def parse(self):
         with open(outputJSON) as jsonFile:
           measurementsJSON = json.load(jsonFile)
           self.readMeasurements(measurementsJSON)
+      else:
```
Since template identifier is optional, can you also add the exception handling?

che85 replied:
I added an exception for that.

fedorov commented:
Yes, I think it should not skip the file. If dcmqi is not available, it still makes sense to extract composite context for the input SRs.

che85 committed:
ENH: removed skipping message if dcmqi was not found

* continue without reading SR measurements

che85 commented:
@fedorov I just removed the "skipping" message. Feel free to test and merge.

che85 committed:
ENH: added TIDNotSupportedError exception

che85 committed:
ENH: handling TIDNotSupportedError

fedorov committed:
ENH: fix few robustness issues

 - fix print call
 - make dcmqi binary locator more robust - account for the situation when the converter is in the system path
